Python tests return a value instead of `None`, causing issues with newer versions of pytest

### Overview of the issue

Since version 8.4.0 of pytest (see [changelog](https://docs.pytest.org/en/stable/changelog.html#pytest-8-4-0-2025-06-02)), returning anything from a function that pytest is testing raises a hard error (see for instance the logs in https://github.com/neuronsimulator/nrn-build-ci/actions/runs/15584191385/job/43886272742):

```plaintext
=================================== FAILURES ===================================
_________________________________ test_spikes __________________________________
Expected None, but test returned <TopLevelHocInterpreter>. Did you mean to use `assert` instead of `return`?
```

This is a bug in NEURON tests since it can lead to unintended behavior (see discussion in https://github.com/pytest-dev/pytest/issues/7337 for why it's a bug).

### Expected result/behavior

The tests should pass without issues.

### NEURON setup
 
 - Version: 8.2.7
 - Installation method: pip
 
### Additional notes

The following tests are affected:

```plaintext
The following tests FAILED:
	 29 - pynrn::basic_tests (Failed)
Errors while running CTest
	 31 - rxdmod_tests::rxd_tests (Failed)
	 32 - rxdmod_tests::rxd_mpi_tests (Failed)
	 42 - coreneuron_modtests::spikes_py_cpu (Failed)
	 43 - coreneuron_modtests::spikes_file_mode_py_cpu (Failed)
	 47 - coreneuron_modtests::test_netmove_py_cpu (Failed)
	 49 - coreneuron_modtests::test_watchrange_py_cpu (Failed)
	 52 - coreneuron_modtests::test_natrans_py_cpu (Failed)
```

The fix is to either:

1. fix all of the failing tests to not return anything
2. pin pytest to version `<8.4.0`, which doesn't produce an error (we use `8.1.1` on `master`)

**UPDATE**: apparently this was partially fixed for `master` in #3114 and #2627, possibly others.

--- a/nmodl_requirements.txt
+++ b/nmodl_requirements.txt
@@ -6,7 +6,7 @@ pyyaml>=3.13,<=6.0.2
 find_libpython<=0.4.0
 sympy>=1.3,<=1.13.3
 # dependencies for test
-pytest<=8.1.1
+pytest<=8.4.0
 pytest-cov
 numpy>=1.9.3,<=2.2.3
 scipy

--- a/packaging/python/test_requirements.txt
+++ b/packaging/python/test_requirements.txt
@@ -1,4 +1,4 @@
-pytest<=8.1.1 # potential bug from 8.2.0 due to parallelism?
+pytest<=8.4.0
 # for coverage
 pytest-cov<=6.0.0
 # for RXD test

--- a/test/rxd/test_savestate_basic.py
+++ b/test/rxd/test_savestate_basic.py
@@ -215,7 +215,7 @@ def test_savestate_basic(neuron_nosave_instance):
         return vecs, vecs_std_save
 
     # run the test
-    return test_savestate_rxd()
+    test_savestate_rxd()
 
 
 if __name__ == "__main__":

